File: lego_be/src/main/java/com/example/book_store_be/service/impl/LegoService.java
package com.example.book_store_be.service.impl;

import com.example.book_store_be.model.Lego;
import com.example.book_store_be.repository.ILegoRepository;
import com.example.book_store_be.service.ILegoService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.math.BigDecimal;
import java.util.List;

@Service
public class LegoService implements ILegoService {

    @Autowired
    private ILegoRepository repository;
    @Override
    public Page<Lego> findAllByNameAndPrice(String name, Pageable pageable ) {
        return repository.findAllByNameAndPrice( name,  pageable);
    }

    @Override
    public Lego findById(Integer id) {
        return repository.findById(id).get();
    }

    @Override
    public List<Lego> findByList() {
        return repository.findAll();
    }
}

File: lego_fe/src/service/legoService.js
import axios from "axios";

export const findByLego = async (name, page) => {
    try {
        return (await axios.get(`http://localhost:8080/lego/list?name=${name ? name : ''}&page=${page ? page : 0}`)).data
    } catch (e) {
        console.log(e)
    }
}

export const findById = async (id) => {
    try {
        return (await axios.get(`http://localhost:8080/lego/${id}`)).data
    } catch (e) {
        console.log(e)
    }
}

export const findByImg = async (id) => {
    try {
        return (await axios.get(`http://localhost:8080/lego/legoImgs/${id}`)).data
    }catch (e){
        console.log(e)
    }
}